Recruit-Training/Basic-Recruit-Training/pointers/main.cpp

```diff
@@ -55,3 +60,10 @@ void var(int *a, int b)
     *a = 365;
     b = 912;
 }
+
+void mySwap(int* a, int* b)
+{
+    *a = *a xor *b;
+    *b = *b xor *a;
+    *a = *a xor *b;
```

Comment:
This is an interesting way to go about it and does work. I've never seen anyone do swaps like this but a quick google search tells me that this is cool but bad practice.  It may be more efficient if you were programming in assembly but modern CPUs can usually handle swaps in a single instruction. The compile might not notice what you're doing and this might end up being slower (3 instructions vs 1). Also, if you were to debug this in half a year that's probably painful. 

More info why this is might be a bad idea:
https://en.wikipedia.org/wiki/XOR_swap_algorithm